core/src/main/java/io/questdb/griffin/engine/functions/window/SumDoubleWindowFunctionFactory.java

```diff
@@ -0,0 +1,757 @@
+
+        private long count = 0;
+        private double externalSum;
+        private double sum = 0.0;
+
+        public SumOverUnboundedRowsFrameFunction(Function arg) {
+            super(arg);
+        }
+
+        @Override
+        public void computeNext(Record record) {
+            double d = arg.getDouble(record);
+            if (Numbers.isFinite(d)) {
+                sum += d;
+                count++;
+            }
+
+            externalSum = count != 0 ? sum : Double.NaN;
+        }
+
+        @Override
+        public double getDouble(Record rec) {
+            return externalSum;
+        }
+
+        @Override
+        public int getPassCount() {
+            return WindowFunction.ZERO_PASS;
+        }
+
+        @Override
+        public void pass1(Record record, long recordOffset, WindowSPI spi) {
+            computeNext(record);
+
+            Unsafe.getUnsafe().putDouble(spi.getAddress(recordOffset, columnIndex), externalSum);
+        }
+
+        @Override
+        public void reset() {
+            super.reset();
+            externalSum = Double.NaN;
+            count = 0;
+            sum = 0.0;
+        }
+
+        @Override
+        public void toPlan(PlanSink sink) {
+            sink.val(NAME);
+            sink.val('(').val(arg).val(')');
+            sink.val(" over (rows between unbounded preceding and current row)");
+        }
+
+        @Override
+        public void toTop() {
+            super.toTop();
+
+            externalSum = Double.NaN;
+            count = 0;
+            sum = 0.0;
+        }
+    }
+
+    // sum() over () - empty clause, no partition by no order by, no frame == default frame
+    static class SumOverWholeResultSetFunction extends BaseSumFunction {
+        private long count;
+        private double externalSum;
+        private double sum;
+
+        public SumOverWholeResultSetFunction(Function arg) {
+            super(arg);
+        }
+
+        @Override
+        public int getPassCount() {
+            return WindowFunction.TWO_PASS;
+        }
+
+        @Override
+        public void pass1(Record record, long recordOffset, WindowSPI spi) {
+            double d = arg.getDouble(record);
+            if (Numbers.isFinite(d)) {
+                sum += d;
+                count++;
+            }
+        }
+
+        @Override
+        public void pass2(Record record, long recordOffset, WindowSPI spi) {
+            Unsafe.getUnsafe().putDouble(spi.getAddress(recordOffset, columnIndex), externalSum);
+        }
+
+        @Override
+        public void preparePass2() {
+            externalSum = count > 0 ? sum : Double.NaN;
+        }
+
+        @Override
+        public void reset() {
+            super.reset();
+            externalSum = Double.NaN;
+            count = 0;
+            sum = 0.0;
+        }
+
+        @Override
+        public void toTop() {
+            super.toTop();
+
+            externalSum = Double.NaN;
```

Comment:
untested

Reply:
what i mean is this:
![image](https://github.com/questdb/questdb/assets/7276403/0a4a7983-6c40-48f9-888b-8cfa477ffd64)